src/algorand_blocks/state_proof_tracking.rs

```diff
@@ -0,0 +1,88 @@
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    algorand_errors::AlgorandError,
+    algorand_types::{Bytes, Result},
+};
+
+fn is_zero(num: &Option<u64>) -> bool {
+    match num {
+        Some(val) => val == &0,
+        None => true,
+    }
+}
+
+fn is_empty_vec<T>(vec: &Option<Vec<T>>) -> bool {
+    match vec {
+        Some(vec) => vec.is_empty(),
+        None => true,
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct StateProofTracking {
+    #[serde(rename = "v", skip_serializing_if = "is_empty_vec")]
+    pub voters_commitment: Option<Bytes>,
+    #[serde(rename(serialize = "t"), skip_serializing_if = "is_zero")]
+    pub online_total_weight: Option<u64>,
+    #[serde(rename = "n")]
+    pub next_round: Option<u64>,
+}
+
+impl StateProofTracking {
+    pub fn from_json(json: &StateProofTrackingJson) -> Result<Self> {
+        Ok(Self {
+            voters_commitment: json
+                .voters_commitment
+                .as_ref()
+                .map(|commitment| commitment.to_vec()),
+            next_round: json.next_round,
+            online_total_weight: json.online_total_weight,
+        })
+    }
+
+    pub fn to_json(&self, ttype: u64) -> StateProofTrackingJson {
```

Comment:
Notice I pass the type here because it became the index in the hash table and it's not there in StateProofTracking struct.